src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -2761,7 +2761,12 @@ int RadosObject::handle_obj_expiry(const DoutPrefixProvider* dpp, optional_yield
           bufferlist bl;
           bl.append(tier_config.name);
           attrs[RGW_ATTR_STORAGE_CLASS] = bl;
-
+	  {
+	    ceph::real_time deletion_time = real_clock::now();
+	    bufferlist bl;
+	    encode(deletion_time, bl);
+	    attrs[RGW_ATTR_INTERNAL_MTIME] = std::move(bl);
+	  }
           const req_context rctx{dpp, y, nullptr};
           return obj_op.write_meta(0, 0, attrs, rctx, head_obj->get_trace());
```

Comment:
Since this expiration is done only for temporarily restored copies ( which are not being synced),  we need not sync these reset_stub changes too. I think we need to pass 'log_op=false' here as well.